conan/tools/files/files.py

```diff
@@ -262,6 +262,25 @@ def chdir(conanfile, newdir):
         os.chdir(old_path)
 
 
+class _ProgressPrinter:
+    def __init__(self, output, uncompressed_size: int, step: int = 1):
+        self.enabled = hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
+        self.uncompressed_size = uncompressed_size
+        self.percentage = 0
+        self.step = step
+        self.output = output
+
+    def print_progress(self, current_size: int):
+        if not self.enabled:
+            return
+        txt_msg = "Unzipping %d %%"
+        current_percentage = int(current_size * 100.0 / self.uncompressed_size) if self.uncompressed_size != 0 else 0
+        if current_percentage >= self.percentage + self.step:
+            if 100 - current_percentage == self.step:
+                current_percentage = 100
+            self.output.rewrite_line(txt_msg % current_percentage)
```

Comment:
Rewrites are not that great, specially for CI output, which result in endless and annoying logs if enabled, and not a trace at all if not enabled. The "timed-output" seems to be a good balance.

Reply:
We were already using `output.rewrite_line` for the unzip progress but I agree it could cause some inconvenience. 
With this refactor, we could remove the `rewrite` by using `FileProgress` class.